spectator/registry.py

```diff
@@ -1,286 +1,79 @@
+        return Timer(self._new_meter(name, tags), clock=self._clock, spectatord=self._spectatord)
+
+    def pct_timer(self, name: str, tags: Optional[dict] = None) -> PercentileTimerV2:
+        return PercentileTimerV2(self._new_meter(name, tags), clock=self._clock,
+                                 spectatord=self._spectatord)
 
     def start(self, config=None):
-        if self._started:
-            logger.debug("registry already started")
-            return RegistryStopper(None)
-        else:
-            self._started = True
-            logger.info("starting registry")
-            if config is None:
-                logger.info("config not specified, using default")
-                config = defaultConfig
-            elif type(config) is not dict:
-                logger.warning("invalid config specified, using default")
-                config = defaultConfig
-            frequency = config.get("frequency", 5.0)
-            self._uri = config.get("uri", None)
-            self._batch_size = config.get("batch_size", 10000)
-            self._common_tags = config.get("common_tags", {})
-            self._client = HttpClient(self, config.get("timeout", 1))
-            self._timer = RegistryTimer(frequency, self._publish)
-            self._timer.start()
-            logger.debug("registry started with config: %s", config)
-            return RegistryStopper(self)
+        """Avoid breaking the API."""
 
     def clear_meters_and_start(self):
-        """
-        This is called after a fork in the child process
-        to clear the cloned `_meters` and prevent duplicates
-        (the `_meters` are copied with the process
-        during the forking)
-        """
-        self._meters = {}
-        self.start()
-
-    def stop_without_publish(self):
-        """
-        This is called before a fork to prevent a potential deadlock.
-        It cancels the background timer thread. After the fork, the timer
-        thread is restarted in the main and cloned processes.
-        """
-        if self._started:
-            logger.debug("stopping log registry")
-            self._timer.cancel()
-            self._started = False
+        """Avoid breaking the API."""
 
     def stop(self):
-        self.stop_without_publish()
-        # Even if not started, attempt to flush data to minimize risk
-        # of data loss
-        self._publish()
-
-    def _get_measurements(self):
-        """
-        If there are no references in user code, then we expect four references to a meter:
+        """Avoid breaking the API."""
 
-            1) meters map,
-            2) local variable in the for loop,
-            3) internal to ref count method, and
-            4) internal to the garbage collector.
-
-        """
-        snapshot = []
+    def stop_without_publish(self):
+        """Avoid breaking the API."""
 
+    def __iter__(self):
+        """Avoid breaking the API."""
         with self._lock:
```

Comment:
Why is the lock still needed?

Reply:
Not needed, since no state is being mutated. We can remove it.